app/models/concerns/slug_bug.rb

```diff
@@ -22,14 +19,15 @@ def set_slug
                 else
                   id
                 end
+    self.slug_for_upgrade = self.slug
   end
 
   private
 
     # Cribbed from https://gitlab.com/notch8/louisville-hyku/-/blob/main/app/models/custom_slugs/slug_behavior.rb#L14
     def remove_index_and_reindex
       # rubocop:disable Rails/Blank
-      return unless slug.present?
+      return unless (slug.present? || slug_for_upgrade.present?)
```

Comment:
just want to check my understanding here.

- existing works will have a `slug` property
- new works will have `slug` and `slug_for_upgrade`

once the migration happens, we want to use `slug_for_upgrade`? and so they check for either here ensures that this method works now and in the future?

Reply:
@jeremyf just want to check my understanding here.

- existing works will have a `slug` property
- new works will have `slug` and `slug_for_upgrade`

once the migration happens, we want to use `slug_for_upgrade`? and so they check for either here ensures that this method works now and in the future?